eval/src/value/lambda.rs

```diff
@@ -0,0 +1,34 @@
+use derive_more::{Debug, Display};
+use parser::ast::Expression;
+use std::rc::Rc;
+
+use crate::{context::Context, errors::EvalError};
+
+use super::JmlValue;
+
+pub type Identifier<'source> = &'source str;
+
+pub type NativeFunction<'source, Span> = dyn Fn(Span, Vec<JmlValue<'source>>, &mut Context<'source>) -> Result<JmlValue<'source>, EvalError>
```

Comment:
If you can make all the JmlValue variants owned here you can keep only Context bound to the 'source lifetime. e.g. 

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=1bef1d887175ac63f858cd3b90c774b1

Reply:
You`re right, in your case code becomes cleaner, but it is planned to use Cow type for String type (to handle strings from the source code and those created in runtime) in the future, so I decided to make such a structure for MVP, the interface of which will not change in the future

Reply:
I see, so you aiming to make it zero copy in the future. Makes sense.